postqa/cli.py

```diff
@@ -15,27 +15,37 @@
 from . import jsonshim
 from . import lsstsw
 from . import jenkinsenv
-from .schemas import validate
+from .schemas import load_schema, validate
 
 
 def run_post_qa():
     """CLI entrypoint for the ``post-qa`` command."""
     args = parse_args()
 
-    job_json = build_job_json(args.qa_json_path, args.lsstsw_dirname,
-                              args.accepted_metrics)
+    registered_metrics = load_registered_metrics(api_url=args.api_url)
 
+    metric_json, job_json = build_json_docs(args.qa_json_path,
+                                            args.lsstsw_dirname,
+                                            registered_metrics)
     if not args.test:
-        upload_json(job_json, api_url=args.api_url,
-                    api_user=args.api_user, api_password=args.api_password)
+
+        upload_json_doc(metric_json, api_url=args.api_url,
```

Comment:
Do we (or should we) skip this upload if there are no new metrics, beyond what's in `registered_metrics`?